### PR TITLE
Gradgradcheck to runs successfully with unrelated inputs

### LINKED ISSUE #57649: gradgradcheck fails if the function does not depend on the input
## 🐛 Bug

If one (or more) of the inputs to `gradgradcheck` has no dependency to the output, it raises an error.
This does not happen with `gradcheck` function.
This might sound silly (e.g. "why would you put unrelated input to gradgradcheck?"), but please take a look at the example below.

## To Reproduce

    import torch

    def fcn0(x, a, b):
        return (x * a) ** 2 + b

    def functional(fcninp, x, a, b):
        f = fcninp(x, a, b)
        return torch.autograd.grad(f, x, create_graph=True)[0]

    x = torch.tensor(1.0, dtype=torch.float64).requires_grad_()
    a = torch.tensor(1.0, dtype=torch.float64).requires_grad_()
    b = torch.tensor(1.0, dtype=torch.float64).requires_grad_()

    # success
    torch.autograd.gradgradcheck(functional, (fcn0, x, a, b.detach()))
    # fail
    torch.autograd.gradgradcheck(functional, (fcn0, x, a, b))

So, `functional` is a functional that takes a function and other parameters as its inputs, and returns the derivative of the function w.r.t. one of the inputs.
If the derivative of the function just happen to have no dependence to one of the parameters, then `gradgradcheck` raises an error.

## Expected behavior

`gradgradcheck` should not raise an error in the above case.

## Environment

 - PyTorch Version (e.g., 1.0): 1.9.0.dev20210419
 - OS (e.g., Linux): WSL Ubuntu 20.04
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): N/A
 - Python version: 3.8.5
 - CUDA/cuDNN version: N/A
 - GPU models and configuration: N/A
 - Any other relevant information: N/A

## Additional context

`gradcheck` is fine if one of the inputs are unrelated to the output, e.g.

    import torch

    def fcn0(x, a):
        return x ** 2

    x = torch.tensor(1.0, dtype=torch.float64).requires_grad_()
    a = torch.tensor(1.0, dtype=torch.float64).requires_grad_()

    torch.autograd.gradcheck(fcn0, (x, a))  # returns True

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano

### PR DESCRIPTION
**Pull request by mfkasim1**
Fixes #57649

### CODE REVIEW

**Review by albanD — APPROVED** [review]
LGTM !
Thanks for the fix!

**Comment by albanD** [comment]
Not sure why mypy is flagging a file that is not touched by this PR...
